containerd/containerd.go

```diff
@@ -93,7 +95,11 @@ func (d *Driver) createContainer(containerConfig *ContainerConfig, config *TaskC
 
 	var opts []oci.SpecOpts
 
-	opts = append(opts, oci.WithImageConfigArgs(containerConfig.Image, args))
+	if config.Entrypoint != "" {
+		opts = append(opts, oci.WithProcessArgs(args...))
```

Comment:
when specified as empty string, docker will ignore the entrypoint in the image in my memorize.

maybe we should compare to nil here?

Reply:
Can you give an example?

Reply:
i do not have laptop nearby, this link may help

https://github.com/moby/moby/pull/23718

Reply:
If I am following you correctly, you are saying if I launch a job with the following config
Here `ENTRYPOINT` defined in the image is `["/bin/echo", "hello"]`

```
      config {
        image = "shm32/entrypoint:hello"
	entrypoint = ""
	args = ["shishir"]
      }
```

It will ignore the `entrypoint` defined in the image. I don't think it will happen with `containerd-driver` since the check itself is `if config.Entrypoint != ""` and it will just skip that.

I tried running this job, and it prints `hello shishir` correctly.

Reply:
the expect behavior should be ignore the entrypoint in the image and just execute the command in args?

Reply:
i mean:

It **should** ignore the entrypoint defined in the image.

sorry for my poor english

Reply:
What does that mean? If you are ignoring the `entrypoint` defined in the image, who are you supplying the args to?
Also, why would you provide the `command` in the `args`?

This is the reason I want to make `entrypoint` as a `string`. We don't **need** to follow `nomad docker driver` 100% and have a 1:1 mapping for each option. `entrypoint` is defined as the executable which will serve as the entry point to the container i.e the first thing that will get executed when the container is launched. Ideally it should be a single executable, and if there are any arguments to the executable that should be supplied separately as part of `args` 

Having `entrypoint` as `string` enforces that! 

I think we need to define the expected behavior 🙂

1. User should not be setting `entrypoint = ""`.
2. If they did (1), I am letting the `containerd-driver` assume, they did it by accident, and fallback on the default `entrypoint` defined in the image.

Reply:
okay I think I see what you are saying 🙂 . There is no way to `ignore` the entrypoint, only to `override` the entrypoint.

Reply:
your point is clear and is good if not consider the compatibility.

i just think more about using the same config between docker driver and containerd driver.

Reply:
so just ignore what i say because you already consider it deeperly

Reply:
Your point about ignoring the `entrypoint` is valid. Currently there is no way to `ignore` the `entrypoint`, one can only `override` the entrypoint.

Like you suggested, I will change the check to `nil` which will allow to ignore the `entrypoint` too.

Reply:
Fixed.